falcon/api.py

```diff
@@ -544,13 +545,13 @@ def _call_req_mw(self, stack, req, resp):
             # Put executed component on the stack
             stack.append(component)  # keep track from outside
 
-    def _call_rsrc_mw(self, stack, req, resp, resource):
+    def _call_rsrc_mw(self, stack, req, resp, resource, params):
         """Run process_resource middleware methods."""
 
         for component in self._middleware:
             _, process_resource, _ = component
             if process_resource is not None:
-                process_resource(req, resp, resource)
+                process_resource(req, resp, resource, params)
```

Comment:
All things considered, I think it's OK to make the breaking change. There will be very few breaking changes in the next release as it is, and this one is easy enough to work around. Let's go ahead and land this for 0.4. That will be the final release before 1.0.